lib/providers/notification/notificationsProvider.tsx

```diff
@@ -0,0 +1,86 @@
+import { v4 as uuid } from 'uuid';
+import React, { useState, PropsWithChildren } from "react";
+import { NotificationsContext } from "./notificationsContext";
+import { NotificationBannerProps } from "@components/Notification/banner";
+import { NotificationItemProps } from "@components/Notification/item";
+import { STATUS } from "@utils/notification-utils";
+
+export const NotificationsProvider = ({ children }: PropsWithChildren) => {
+    const [banners, setBanners] = useState<
+        Array<NotificationBannerProps & { id: string }>
+    >([]);
+    const [notificationItems, setNotificationItems] = useState<
+        Array<NotificationItemProps & { id: string }>
+    >([]);
+
+    const addBanner = (props: Omit<NotificationBannerProps, "id">) => {
+        setBanners((prev) => [...prev, { ...props, id: uuid() }]);
+    };
```

Comment:
Since there are a lot of actions, should we use useReducer? We could also use different contexts for actions and values to help the app perform better as it grows, especially since this provider covers the whole app.

Reply:
this provider is simply supposed to be used in the external app by wrapping it but it doesn't cover the whole app's state. It won't be extended any further because it exists to cover the needs of Notifications components only.
This provider was added based on this request: https://github.com/deriv-com/quill-ui/pull/73#discussion_r1577141973, same as in this PR: https://github.com/deriv-com/quill-ui/pull/73 for another component - snackbar.
When we use all these providers for snackbar and for notifications, we can make it clean just like how we have on flutter - https://dev.to/hyposlasher/no-more-react-context-hell-4pm3